apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx

```diff
@@ -1,25 +1,129 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+import { useWallet } from '@mysten/wallet-adapter-react';
+import { WalletWrapper } from '@mysten/wallet-adapter-react-ui';
+import clsx from 'clsx';
+import { useMemo } from 'react';
+import { useFieldArray } from 'react-hook-form';
+import toast from 'react-hot-toast';
+import { z } from 'zod';
+
+import { FunctionParamInput } from './FuncitonParamInput';
+import { useFunctionParamsDetails } from './useFunctionParamsDetails';
+
 import type { SuiMoveNormalizedFunction, ObjectId } from '@mysten/sui.js';
 
+import { useZodForm } from '~/hooks/useZodForm';
+import { Button } from '~/ui/Button';
 import { DisclosureBox } from '~/ui/DisclosureBox';
 
+const argsSchema = z.object({
+    params: z.array(z.object({ value: z.string().trim().min(1) })),
+});
+
 export type ModuleFunctionProps = {
     packageId: ObjectId;
     moduleName: string;
     functionName: string;
     functionDetails: SuiMoveNormalizedFunction;
     defaultOpen?: boolean;
 };
+
 export function ModuleFunction({
     defaultOpen,
+    packageId,
+    moduleName,
     functionName,
     functionDetails,
 }: ModuleFunctionProps) {
+    const { connected, signAndExecuteTransaction } = useWallet();
+    const allParamsDetails = useFunctionParamsDetails(
+        functionDetails.parameters
+    );
+    const filteredParamsDetails = useMemo(
+        () => allParamsDetails.filter((aParam) => !aParam.isTxContext),
+        [allParamsDetails]
+    );
+    const defaultValues = useMemo(
+        () =>
+            Array.from({ length: filteredParamsDetails.length }, () => ({
+                value: '',
+            })),
+        [filteredParamsDetails.length]
+    );
+    const { register, handleSubmit, formState, control } = useZodForm(
+        argsSchema,
+        {
+            defaultValues: {
```

Comment:
Given the initial value is an empty string, do we even need default values?

Reply:
this is done mainly to initialise `fields` with the correct number of elements. (I think also show somewhere it doc it good to always define default values.)

I will remove the `useFieldArray` so it wont be necessary anyway.